tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,170 @@
+
+    def TestRemoveImage(self):
+        """
+        Test remove image and remove snapshot
+        """
+        isRemoved = self.fs.remove(CEPH_IMG)
+        self.assertTrue(isRemoved)
+        imgs = self.fs.list_images()
+        self.assertNotIn(CEPH_IMG, imgs)
+
+    def runTest(self):
+        self.TestGetImage()
+        self.TestListImages()
+        self.TestWrite()
+
+    def tearDown(self):
+        self.TestRemoveImage()
+
+
+class TestSnapshot(unittest.TestCase):
+    """
+    create snapshot
+    """
+    @trace
+    def setUp(self):
+        self.NOT_EXIST_SNAP_NAME = "TEST_SNAP_NOT_FOUND"
+        self.fs = ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE, False, 1)
+
+    def TestSnapImage(self):
+        """
+        Test create snapshot
+        """
+        snapFlag = self.fs.snap_image(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(snapFlag)
+
+    def TestListSnapshot(self):
+        """
+        Test list snapshot
+        """
+        snapList = self.fs.list_snapshots(CEPH_IMG)
+        self.assertIn(CEPH_SNAP_IMG, snapList)
+        self.assertNotIn(self.NOT_EXIST_SNAP_NAME, snapList)
+
+    def TestSnapProtect(self):
+        """
+        Test protecting a snapshot
+        """
+        protectFlag = self.fs.snap_protect(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(protectFlag, "Failed to protect a snapshot!")
+
+    def TestSnapUnprotect(self):
+        """
+        Test unprotecting a snapshot
+        """
+        unprotectFlag = self.fs.snap_unprotect(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(unprotectFlag, "Failed to unprotect a snapshot")
+
+    def TestClone(self):
+        """
+        Test clone image
+        """
+        self.TestSnapProtect()
+        cloneFlag = self.fs.clone(CEPH_IMG, CEPH_SNAP_IMG,
+                                  CEPH_CHILD_IMG)
+        self.assertTrue(cloneFlag, "Failed to clone image")
+        children = self.fs.list_children(CEPH_IMG, CEPH_SNAP_IMG)
+        children_imgs = [child[1] for child in children if child[0] == POOL]
+        self.assertIn(CEPH_CHILD_IMG, children_imgs,
+                      "Error, cannot find child image")
+        parent_info = self.fs.get_parent_info(CEPH_CHILD_IMG)
+        self.assertEqual(CEPH_IMG, parent_info[1])
+        self.assertEqual(CEPH_SNAP_IMG, parent_info[2])
+        self.fs.flatten(CEPH_CHILD_IMG)
+        self.TestSnapUnprotect()
+
+    def TestRemoveSnapshot(self):
+        """
+        Test Remove Snapshot
+        """
+        self.assertRaises(file_system_exceptions.ImageHasSnapshotException,
+                          self.fs.remove, CEPH_IMG)
+        removeSnapFlag = self.fs.remove_snapshot(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(removeSnapFlag, "Failed to remove a snapshot")
+
+    def runTest(self):
+        self.TestSnapImage()
+        self.TestListSnapshot()
+        self.TestClone()
+        self.TestRemoveSnapshot()
```

Comment:
Remove this entire function.
The unitest framework will call each of your test function with setup and teardown before and after each function.